src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.Cci;
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.InteropServices;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a compiler generated and embedded attribute type.
+    /// This type has the following properties:
+    /// 1) It is non-generic, sealed, internal, non-static class.
+    /// 2) It derives from System.Attribute
+    /// 3) It has Microsoft.CodeAnalysis.EmbdeddedAttribute
+    /// 4) It has System.Runtime.CompilerServices.CompilerGeneratedAttribute
+    /// 5) It has a parameter-less constructor
+    /// </summary>
+    internal sealed class SynthesizedEmbeddedAttributeSymbol : NamedTypeSymbol
+    {
+        private readonly string _name;
+        private readonly NamedTypeSymbol _baseType;
+        private readonly NamespaceSymbol _namespace;
+        private readonly ImmutableArray<Symbol> _members;
+        private readonly ModuleSymbol _module;
+
+        public SynthesizedEmbeddedAttributeSymbol(AttributeDescription description, CSharpCompilation compilation, DiagnosticBag diagnostics)
+        {
+            _name = description.Name;
+            _baseType = compilation.GetWellKnownType(WellKnownType.System_Attribute);
+
+            // Report errors in case base type was missing (MissingMetadataTypeSymbol)
```

Comment:
>was missing (MissingMetadataTypeSymbol) [](start = 47, length = 39)

"is missing or bad"? A type doesn't have to be missing to have an use-site diagnostics.